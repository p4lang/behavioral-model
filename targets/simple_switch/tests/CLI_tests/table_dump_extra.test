#!/bin/sh
./run_one_test.py $srcdir/testdata table_dump_extra table_dump_extra.json
