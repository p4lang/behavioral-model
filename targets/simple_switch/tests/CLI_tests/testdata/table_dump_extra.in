table_dump empty_key
table_set_default empty_key _nop
table_dump empty_key
table_reset_default empty_key
table_dump empty_key
