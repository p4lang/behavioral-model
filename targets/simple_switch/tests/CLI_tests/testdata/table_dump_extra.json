{
    "header_types": [
        {
            "name": "standard_metadata_t",
            "id": 0,
            "fields": [
                [
                    "ingress_port",
                    9
                ],
                [
                    "packet_length",
                    32
                ],
                [
                    "egress_spec",
                    9
                ],
                [
                    "egress_port",
                    9
                ],
                [
                    "egress_instance",
                    32
                ],
                [
                    "instance_type",
                    32
                ],
                [
                    "clone_spec",
                    32
                ],
                [
                    "_padding",
                    5
                ]
            ],
            "length_exp": null,
            "max_length": null
        }
    ],
    "headers": [
        {
            "name": "standard_metadata",
            "id": 0,
            "header_type": "standard_metadata_t",
            "metadata": true
        }
    ],
    "header_stacks": [],
    "parsers": [
        {
            "name": "parser",
            "id": 0,
            "init_state": "start",
            "parse_states": [
                {
                    "name": "start",
                    "id": 0,
                    "parser_ops": [],
                    "transition_key": [],
                    "transitions": [
                        {
                            "value": "default",
                            "mask": null,
                            "next_state": null
                        }
                    ]
                }
            ]
        }
    ],
    "deparsers": [
        {
            "name": "deparser",
            "id": 0,
            "order": []
        }
    ],
    "meter_arrays": [],
    "actions": [
        {
            "name": "_nop",
            "id": 0,
            "runtime_data": [],
            "primitives": []
        }
    ],
    "pipelines": [
        {
            "name": "ingress",
            "id": 0,
            "init_table": "empty_key",
            "tables": [
                {
                    "name": "empty_key",
                    "id": 0,
                    "match_type": "exact",
                    "type": "simple",
                    "max_size": 16384,
                    "with_counters": false,
                    "direct_meters": null,
                    "support_timeout": false,
                    "key": [],
                    "actions": [
                        "_nop"
                    ],
                    "next_tables": {
                        "_nop": null
                    },
                    "base_default_next": null
                }
            ],
            "conditionals": []
        },
        {
            "name": "egress",
            "id": 1,
            "init_table": null,
            "tables": [],
            "conditionals": []
        }
    ],
    "calculations": [],
    "checksums": [],
    "learn_lists": [],
    "field_lists": [],
    "counter_arrays": [],
    "register_arrays": [],
    "force_arith": [
        [
            "standard_metadata",
            "ingress_port"
        ],
        [
            "standard_metadata",
            "packet_length"
        ],
        [
            "standard_metadata",
            "egress_spec"
        ],
        [
            "standard_metadata",
            "egress_port"
        ],
        [
            "standard_metadata",
            "egress_instance"
        ],
        [
            "standard_metadata",
            "instance_type"
        ],
        [
            "standard_metadata",
            "clone_spec"
        ],
        [
            "standard_metadata",
            "_padding"
        ]
    ]
}