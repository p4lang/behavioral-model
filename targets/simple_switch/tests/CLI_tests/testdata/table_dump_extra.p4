parser start {
    return ingress;
}

action _nop() { }

table empty_key {
    actions { _nop; }
}

control ingress {
    apply(empty_key);
}
