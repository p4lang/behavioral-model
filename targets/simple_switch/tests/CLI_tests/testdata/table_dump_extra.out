????
==========
TABLE ENTRIES
==========
Dumping default entry
EMPTY
==========
????
Setting default action of empty_key
action:              _nop
runtime data:        
????
==========
TABLE ENTRIES
==========
Dumping default entry
Action entry: _nop - 
==========
????
????
==========
TABLE ENTRIES
==========
Dumping default entry
EMPTY
==========
????
